addons/addon-base-raas/packages/base-raas-services/lib/data-egress/__test__/data-egress-service.test.js

```diff
@@ -285,11 +295,156 @@ describe('DataEgressService', () => {
+      dataEgressService._settings = {
+        get: settingName => {
+          if (settingName === 'enableEgressStore') {
+            return true;
+          }
+          if (settingName === 'egressStoreKmsKeyAliasArn') {
+            return 'test-egressStoreKmsKeyAliasArn';
+          }
+          if (settingName === 'egressStoreBucketName') {
+            return 'test-egressStoreBucketName';
+          }
+          return undefined;
+        },
+      };
+
+      dbService.table.scan.mockImplementationOnce(() => {
+        throw new Error();
+      });
+      const requestContext = {};
+      const envId = 'test-id';
+
+      await expect(dataEgressService.terminateEgressStore(requestContext, envId)).rejects.toThrow(
+        // It is better to check using boom.code instead of just the actual string, unless
+        // there are a few errors with the exact same boom code but different messages.
+        // Note: if you encounter a case where a service is throwing exceptions with the
+        // same code but different messages (to address different scenarios), you might
+        // want to suggest to the service author to use different codes.
+        expect.objectContaining({ boom: true, code: 'notFound', safe: true }),
+      );
+    });
+
+    it('should fail delete egress store while egress store is in PROCESSING status', async () => {
+      dataEgressService._settings = {
+        get: settingName => {
+          if (settingName === 'enableEgressStore') {
+            return true;
+          }
+          if (settingName === 'egressStoreKmsKeyAliasArn') {
+            return 'test-egressStoreKmsKeyAliasArn';
+          }
+          if (settingName === 'egressStoreBucketName') {
+            return 'test-egressStoreBucketName';
+          }
+          return undefined;
+        },
+      };
+
+      dbService.table.scan.mockResolvedValue([
+        {
+          status: 'PROCESSING',
+          workspaceId: 'test-workspace-id',
+          s3BucketName: 'test-s3BucketName',
+          s3BucketPath: 'test-s3BucketPath',
+          id: 'test-egress-store-id',
+        },
+      ]);
+      const requestContext = {};
+      const envId = 'test-workspace-id';
+
+      await expect(dataEgressService.terminateEgressStore(requestContext, envId)).rejects.toThrow(
+        // It is better to check using boom.code instead of just the actual string, unless
+        // there are a few errors with the exact same boom code but different messages.
+        // Note: if you encounter a case where a service is throwing exceptions with the
+        // same code but different messages (to address different scenarios), you might
+        // want to suggest to the service author to use different codes.
+        expect.objectContaining({ boom: true, code: 'forbidden', safe: true }),
+      );
+    });
+
+    it('should success delete egress store while egress store is not in PROCESSING status', async () => {
```

Comment:
Can we check as an assertion that we are removing egress policy for the specific environment in request? Perhaps add test for `removeEgressStoreBucketPolicy` separately and mock `removeEgressStoreBucketPolicy` in this test and verify we called with right params?

Reply:
Fix applied in commit: 3917dc0ed4b0fa6bb7aafbd67d45cd7846a58922